### DISCUSSION TITLE
Error ketika coding 'length'

### DISCUSSION THREAD

**Discussion by Saupia** [teamalgoritma/community#389] [Q&A] [answered]
Di dalam buku panduan belajar, saya temukan ini:
```
names(vector2) <- c("User ID", "Active", "Cart Items", "Payment")
length(vector2)
```
dan hasilnya
#> [1] 4
--------

Saya coba copy coding yang sama ke dalam R punya saya dan running tetapi ada pembritahuan error. Kodenya masih sama
```{r}
names(vector2) <- c("User ID", "Active", "Cart Items", "Payment")
length(vector2)
```
Ini tampilan pesan errornya

Error in names(vector2) <- c("User ID", "Active", "Cart Items", "Payment") : 
  'names' attribute [4] must be the same length as the vector [3]

Mohon bantuannya. 

Terima kasih.

**Comment by onlyphantom** [comment, marked as answer]
The reason is because `vector2` is a vector with 3 elements. You can verify it with `length(vector2)`. 

So to apply names to this vector, you also need `vector2` to be a vector of 4:
```
vector4 <- c(100, 200, 300, 400)
names(vector4) <- c("User ID", "Active", "Cart Items", "Payment")

# or:
vector3 <- c(100, 200, 300)
names(vector3) <- c("Active", "Cart Items", "Payment")
```

**Comment by Saupia** [comment]
Thank you, Sam,
Understood.

Pada tanggal Rab, 5 Okt 2022 pukul 16.25 Samuel Chan <
***@***.***> menulis:

> The reason is because vector2 is a vector with 3 elements. You can verify
> it with length(vector2).
>
> So to apply names to this vector, you also need vector2 to be a vector of
> 4:
>
> vector4 <- c(100, 200, 300, 400)
> names(vector3) <- c("User ID", "Active", "Cart Items", "Payment")
>
> # or:
> vector3 <- c(100, 200, 300)
> names(vector3) <- c(""Active", "Cart Items", "Payment")
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/teamalgoritma/community/discussions/389#discussioncomment-3803262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3MRF5XQH3EGO25OCTCNOO3WBVCP3ANCNFSM6AAAAAAQ5J7T4A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regards,

*Renhard Saupia*
Jakarta, Indonesia
